edbdeploy/projects/virtualbox.py

```diff
@@ -141,7 +126,49 @@ def provision(self, env):
                 self.ansible_vars['reference_architecture']
 
             # Load specifications
-            env.cloud_spec = self._load_cloud_specs(env)
+            cloud_spec = self._load_cloud_specs(env)
+            defaults = default(cloud_spec)
+
+            if os.path.exists(os.path.join(self.project_path, "spec.json")):
+                user_spec = self._load_spec_file(os.path.join(self.project_path,
+                                                 "spec.json"))
+            else:
+                user_spec = default(VirtualBoxSpec.get(env.reference_architecture))
+
+            # Generate dbt-2 client and driver specs on the fly as it depends on
+            # how many of each are desired.
+            if type(user_spec['dbt2_client']['count']) is int:
```

Comment:
I'd make sure `dbt2_client` exists in `user_spec` before trying to access sub-elements.